app/services/order_detail_batch_updater.rb

```diff
@@ -148,7 +148,13 @@ def update_all_order_status_ids
         end
       # cancel other orders or change status of any order
       else
-        order_detail.change_status!(order_status)
+        order_detail.change_status!(order_status) do |od|
+          # Ideally this would be part of the change_status! method
```

Comment:
does this need to happen in the block?  
I think I'd prefer setting this before the `change_status!` call if it works:
```
order_detail.canceled_by_user = @user
order_detail.canceled_at = Time.current
order_detail.change_status!(order_status)
```